java/feathub-udf/flink-udf/src/main/java/com/alibaba/feathub/flink/udf/processfunction/SlidingWindowKeyedProcessFunction.java

```diff
@@ -168,16 +209,18 @@ public void onTimer(
             outputRow.setField(keyFieldNames[i], ctx.getCurrentKey().getField(i));
         }
 
-        Row rowToAdd = state.timestampToRow.get(timestamp);
         final Row accumulatorStates = state.getAccumulatorStates();
         final List<Integer> leftIdxList = state.getLeftTimestampIdx();
 
         for (AggregationFieldsDescriptor.AggregationFieldDescriptor descriptor :
                 aggregationFieldsDescriptor.getAggFieldDescriptors()) {
             final int aggFieldIdx = aggregationFieldsDescriptor.getAggFieldIdx(descriptor);
             Object accumulatorState = accumulatorStates.getField(aggFieldIdx);
-            if (rowToAdd != null) {
-                descriptor.aggFunc.merge(accumulatorState, rowToAdd.getField(descriptor.fieldName));
+            if (state.timestampToRow.contains(timestamp)) {
+                for (Row rowToAdd : state.timestampToRow.get(timestamp)) {
```

Comment:
I believe that we can get the rows to add once outside of the loop to avoid accessing the state multiple times.